packages/vite/src/node/server/middlewares/proxy.ts

```diff
@@ -126,8 +126,8 @@ export function proxyMiddleware(
             debug(`bypass: ${req.url} use modified options: %O`, options)
             return next()
           } else if (bypassResult === false) {
-            debug(`bypass: ${req.url} -> 404`)
-            return res.end(404)
+            debug(`bypass: ${req.url}`)
+            return next()
```

Comment:
> webpack-dev-server style bypass function
> https://github.com/vitejs/vite/pull/10912/files#diff-b46b4f407fe94435eeb7d1c687d762435fb90d12208a4a1eb7b4883b947b29c2R23

As it's documented here, I think this should return 404 instead of going through other middlewares.
`false` means returning 404 ([Webpack's document](https://webpack.js.org/configuration/dev-server/#:~:text=Return%20false%20to%20produce%20a%20404%20error%20for%20the%20request.)).

Reply:
The information in the webpack documentation is inaccurate

https://github.com/webpack/webpack-dev-server/blob/b0a35f82a282997b305805951c6f19ce4180808c/lib/Server.js#L2272-L2285

`False` means "use next handlers". If neither of them takes action, the default handler will report 404.

Reply:
Ah, I see. From what I can see from that code, we probably need `req.url = null;` before calling `next()`.

> Note that the current implementation in vite generates 500 , not 404.

Just in case, I agree that the current implementation is not correct.

Reply:
Agree to that we can go with `req.url = null` if that's what webpack does too.